test/rules/do_not_catch_errors.dart

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N do_not_catch_errors`
+
+void bad() {
+  try {} on A catch (e) { // LINT
+    // ignore
+  } on B catch (e) { // LINT
```

Comment:
Add the following clause to this `try` statement:
```
} on C { // LINT
  // ignore
```
and the following declaration:
```
class C extends Error {}
```